File: src/PlasticisingTile.API/DTO/Plasticising/PlasticisingTileConfigurationDto.cs
using PlasticisingTile.API.DTO.Interfaces;
using PlasticisingTile.API.DTO.Shared;
using PlasticisingTile.Core.Enums;

namespace PlasticisingTile.API.DTO.Plasticising;

public class PlasticisingTileConfigurationDto : IDto
{
    public DateTimeRangeFilterDto? DateTimeRangeFilter { get; set; }
    public IEnumerable<DatasourceColumnDto> AvailableColumns { get; set; } = new List<DatasourceColumnDto>();
    public IEnumerable<DatasourceColumnDto> SelectedColumns { get; set; } = new List<DatasourceColumnDto>();
    public IEnumerable<PlasticisingTileAggregationEnum> SelectedAggregations { get; set; } = new List<PlasticisingTileAggregationEnum>();
}

File: src/plasticising-tile-webapp/src/app/api/models/plasticising-tile-configuration-dto.ts
/* tslint:disable */
/* eslint-disable */
import { DatasourceColumnDto } from './datasource-column-dto';
import { DateTimeRangeFilterDto } from './date-time-range-filter-dto';
import { PlasticisingTileAggregationEnum } from './plasticising-tile-aggregation-enum';
export interface PlasticisingTileConfigurationDto {
  availableColumns?: null | Array<DatasourceColumnDto>;
  dateTimeRangeFilter?: DateTimeRangeFilterDto;
  selectedAggregations?: null | Array<PlasticisingTileAggregationEnum>;
  selectedColumns?: null | Array<DatasourceColumnDto>;
}
